bgpd/bgp_ecommunity.c

```diff
@@ -367,10 +367,15 @@ static void ecommunity_color_str(char *buf, size_t bufsz, uint8_t *ptr)
 	 *  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 	 */
 	uint32_t colorid;
+	uint8_t color_type;
+	/* get the color type */
+	ptr++;
+	color_type = (*ptr) >> 6;
```

Comment:
Please add the references in the comments about this and where they come from to allow people to find the information faster.